cql3/expr/expression.hh

```diff
@@ -80,11 +81,18 @@ class column_mutation_attribute;
 class function_call;
 class cast;
 class field_selection;
+struct null;
+struct bind_variable;
+struct untyped_constant;
+struct tuple_constructor;
+struct collection_constructor;
+struct usertype_constructor;
 
 /// A CQL expression -- union of all possible expression types.  bool means a Boolean constant.
 using expression = std::variant<bool, conjunction, binary_operator, column_value, column_value_tuple, token,
                                 unresolved_identifier, column_mutation_attribute, function_call, cast,
-                                field_selection>;
+                                field_selection, null, bind_variable, untyped_constant,
+                                tuple_constructor, collection_constructor, usertype_constructor>;
```

Comment:
I wonder if our hierarchy is too flat.  The only common ancestor for, eg, tuple and list is the root?

Reply:
I don't think we'll have cases where we want to specially treat *_constructor relative to the others. They're just expressions.

Reply:
And yet all the visitors added in this PR have identical code for handling tuple_constructor and collection_constructor.  OTOH, we can always add layers in the future, so this is fine for now.